src/CodeGen_Internal.cpp

```diff
@@ -388,6 +388,42 @@ Expr lower_int_uint_mod(const Expr &a, const Expr &b) {
     }
 }
 
+Expr unsigned_long_div(Expr num, const Expr &den, const Scope<Interval> &bounds) {
```

Comment:
I think it would be better to make the upper bound of the result a parameter and pass that from the calling code, rather than requiring Scope<Interval> as a parameter.

Reply:
Done